File: frontend/src/services/interfaces/task.dto.ts
export interface TaskDTO {
    id?: number
    title: string
    supportType: string
    isOnline: boolean
    description: string
    date: string
    time: string
    city: string
    state: string
    status: string
    requestBy: {
        id: number
        firstName?: string
        lastName?: string
        role: string | undefined
    },
    volunteer?: null | {
        id: number
        firstName: string
        lastName: string
    },
    assessment?: null
}

File: backend/src/main/java/com/cuidarmais/demo/DTO/TaskDTO/TaskDTO.java
package com.cuidarmais.demo.DTO.TaskDTO;

import java.time.LocalDate;
import java.time.LocalTime;

import com.cuidarmais.demo.Entities.EntityObjects.Enums.Status;
import com.cuidarmais.demo.Entities.EntityObjects.Enums.SupportType;

public record TaskDTO (
    Long id,
    String title,
    SupportType supportType,
    boolean isOnline,
    String description,
    LocalDate date,
    LocalTime time,
    String city,
    String state,
    Status status,
    UserTaskDTO requestBy,
    UserTaskDTO volunteer
) {
    
}
